qcodes/instrument_drivers/ZI/private/ZILI_generic.py

```diff
@@ -0,0 +1,2123 @@
+            inputselect['Demod {} R'.format(demod)] = 47+demod
+            inputselect['Demod {} Phase'.format(demod)] = 63+demod
+
+        for channel in range(1,3):
+            self.add_parameter('scope_channel{}_input'.format(channel),
+                            label=("Scope's channel {}".format(channel) +
+                                   " input source"),
+                            set_cmd=partial(self._scope_setter, 0, 0,
+                                            ('channels/{}/'.format(channel-1) +
+                                             'inputselect')),
+                            get_cmd=partial(self._getter, 'scopes', 0, 0,
+                                            ('channels/{}/'.format(channel-1) +
+                                             'inputselect')),
+                            val_mapping=inputselect,
+                            vals=vals.Enum(*list(inputselect.keys()))
+                            )
+
+        self.add_parameter('scope_average_weight',
+                            label="Scope Averages",
+                            set_cmd=partial(self._scope_setter, 1, 0,
+                                            'averager/weight'),
+                            get_cmd=partial(self._scope_getter,
+                                            'averager/weight'),
+                            vals=vals.Numbers(min_value=1)
+                            )
+
+        self.add_parameter('scope_trig_enable',
+                            label="Enable triggering for scope readout",
+                            set_cmd=partial(self._setter, 'scopes', 0,
+                                            0, 'trigenable'),
+                            get_cmd=partial(self._getter, 'scopes', 0,
+                                            0, 'trigenable'),
+                            val_mapping={'ON': 1, 'OFF': 0},
+                            vals=vals.Enum('ON', 'OFF')
+                            )
+
+        self.add_parameter('scope_trig_signal',
+                            label="Trigger signal source",
+                            set_cmd=partial(self._setter, 'scopes', 0,
+                                            0, 'trigchannel'),
+                            get_cmd=partial(self._getter, 'scopes', 0,
+                                            0, 'trigchannel'),
+                            val_mapping=inputselect,
+                            vals=vals.Enum(*list(inputselect.keys()))
+                            )
+
+        slopes = {'None': 0, 'Rise': 1, 'Fall': 2, 'Both': 3}
+
+        self.add_parameter('scope_trig_slope',
+                            label="Scope's triggering slope",
+                            set_cmd=partial(self._setter, 'scopes', 0,
+                                            0, 'trigslope'),
+                            get_cmd=partial(self._getter, 'scopes', 0,
+                                            0, 'trigslope'),
+     
```

Comment:
could you go thorugh all the TODO that you have in this PR and make sure that they are still relevant? if you are a kind of person who likes to leave them in the code, perhaps, could you make them more explicit sometimes (for example, on this line, what kind of validation you are talking about, and please try to avoid weak language (like "seems") - these notes only have value if they are 100% correct.